sklearn/linear_model/quantile.py

```diff
@@ -0,0 +1,314 @@
+        else:
+            if self.fit_intercept:
+                parameters = np.zeros(X.shape[1] + 1)
+            else:
+                parameters = np.zeros(X.shape[1] + 0)
+
+        # solve sequence of optimization problems
+        # with different smoothing parameter
+        total_iter = []
+        loss_args = (X, y, self.quantile, self.alpha, self.l1_ratio,
+                     self.fit_intercept, sample_weight)
+        for i in range(self.n_gamma_decreases):
+            gamma = self.gamma * self.gamma_decrease ** i
+            result = optimize.minimize(
+                _smooth_quantile_loss_and_gradient,
+                parameters,
+                args=loss_args + (gamma, ),
+                method='L-BFGS-B',
+                jac=True,
+                options={
+                    'gtol': self.gtol,
+                    'maxiter': self.max_iter - sum(total_iter),
+                }
+                )
+            total_iter.append(result['nit'])
+            prev_parameters = parameters
+            parameters = result['x']
+
+            # for lasso, replace parameters with exact zero,
+            # if this decreases the cost function
+            if self.alpha * self.l1_ratio > 0:
+                value, _ = _smooth_quantile_loss_and_gradient(parameters,
+                                                              *loss_args,
+                                                              gamma=0)
+                for j in range(len(parameters)):
+                    new_parameters = parameters.copy()
+                    old_param = new_parameters[j]
+                    new_parameters[j] = 0
+                    new_value, _ = _smooth_quantile_loss_and_gradient(
+                        new_parameters, *loss_args, gamma=0)
+                    # check if the cost function decreases,
+                    # or increases, but by little, and param is small anyway
+                    if new_value <= value \
+                            or np.abs(old_param) < self.xtol \
+                            and new_value < value + self.gtol:
+                        value = new_value
+                        parameters = new_parameters
+
+            # stop if solution does not change between subproblems
+            if np.linalg.norm(prev_parameters-parameters) < self.xtol:
+                break
+            # stop if maximum number of iterations is exceeded
+            if sum(total_iter) >= self.max_iter:
+                break
+            # stop if gamma is already zero
+            if gamma == 0:
+                break
+        # do I really need to issue this warning?
+        # Its reason is lineSearchError, which cannot be easily fixed
+        if not result['success']:
+            warnings.warn("QuantileRegressor did not converge:" +
+                          " Scipy solver terminated with '%s'."
+                          % str(result['message'])
+                          )
```

Comment:
if you merge master in you could use `utils.optimize._check_optimize_result('lbfgs', result)`.